design/cstor/v1/pool_expansion_by_expanding_disk.md

```diff
@@ -0,0 +1,135 @@
+
+### Non-Goals
+
+- If user dedicated the first part of partitioned disk for cStor and second part for third-party then pool expansion can't be performed   automatically untill and unless user extend the first partition.
+
+## Proposal
+
+### User Stories
+
+As a OpenEBS user I should be able to expand the pool when underlying disk got expanded.
+
+### Current Implementation
+
+- When user creates the CSPC, CSPC controller which is watching for CSPC will get create request and creates cstorpoolinstances and cstorpooldeployments. CstorPoolInstance is responsible for holding the blockdevice information and other pool configurations where as cStor pool deployment have 3 containers cstor-pool-mgmt, cstor-pool and m-exporter as name conveyes cstor-pool container creates pool(ZFS file system) on top of the blockdevices, where as cstor-pool-mgmt sidecar container is responsible for managing these pool. If any changes required as of day 2 operations on the pool then pool configurations will be updated by adding BlockDevices, adding RaidGroup and replacing blockdevices cstor-pool-mgmt looks for any pending pool operations for CStorPoolInstance then performs pending operation on the pool.
+
+Below are high level info to explain the Day2 operations that are currently supported.
+
+- Pool Expansion When BlockDevice/raid group added:
+	cstor-pool-mgmt will iterate over all the BlockDevices that exists in CStorPoolInstance resource and verify whether the corresponding BlockDevice/raid group is part of the pool. If that BlockDevice/raid group is not part of the pool then pool expansion will be triggered on newly added BlockDevice/raid group.
+
+- BlockDevice Replacement In Pool:
+    cstor-pool manager will iterate over all the BlockDevices present in CstorPoolInstance and verifies whether the current blockdevice is for BlockDevice replacement request[How cstor-pool-mgmt can know? We are maintaining annotation called openebs.io/predecessor: <replaced_blockdevice_name> on the claim of that blockdevice]? If requested BlockDevice is valid for replacement then the pool manager will perform BlockDevice replacement operation on the pool.
+
+### Proposed Implementations
+
+- CStor-pool Manger inbuilt CSPI controller watches for CSPI resources and tries to get the desired state via every sync operation. As part of reconciliation logic it will iterate over all the BlockDevices that exist on CStorPoolInstance resource if it found BlockDevice resource capacity(considerable changes) is greater than the existing capacity of BlockDevice on CstorPoolInstanceBlockDevice then we will follow below-mentioned steps to perform the [pool expansion](#Pool_Expansion_Steps).
+- [What if cStor is consuming part of partitioned disk/entier partitioned disk? User has to extend the partition manualy after that NDM will update the capacity on the corresponding blockdevice CR. Once NDM updates the capacity on BlockDevice in next reconciliation of CSPI Pool Expansion will be triggered].
```

Comment:
spelling mistakes `entier` to entire.. `manualy` to manually
Please check for spelling mistakes at other places also